code/game/objects/items/tools/medical_tools.dm

```diff
@@ -0,0 +1,246 @@
+	var/brute = TRUE
+	var/heal_rate = 1.5 //yes better than advanced kits
+	gelefficiency = 56 //same as 5 kits with small cell. same 15 with regular cell
+
+/obj/item/tool/geltool/gadget/kitprinter/update_icon()
+	..()
+	if(!cell || cell.gelquantity <= 0)
+		icon_state = "synthkitoff"
+		return
+	if(brute)
+		icon_state = "synthkitbrute"
+		return
+	icon_state = "synthkitburn"
+
+
+/obj/item/tool/geltool/gadget/kitprinter/proc/update_kitprinter(mob/user, silent=FALSE)
+	if(!cell || cell.gelquantity <= 0)
+		icon_state = "synthkitoff"
+		playsound(loc, 'sound/weapons/saberoff.ogg', 0.25, 1, 2)
+		playsound(loc, 'sound/weapons/guns/misc/empty_alarm.ogg', 25, 1)
+		to_chat(user, "<span class='warning'>The tool beeps as it runs out of gel!</span>")
+		return
+
+	update_icon()
+
+/obj/item/tool/geltool/gadget/kitprinter/examine(mob/user)
+	..()
+	if(cell)
+		to_chat(user, "The gel gauge measures: [cell.gelquantity]/[cell.gelcapacity]</b>")
+	else
+		to_chat(user, "<span class='warning'>It does not have a gel vial in it!</span>")
+
+/obj/item/tool/geltool/gadget/kitprinter/attack_self(mob/user)
+	toggle(user)
+
+/obj/item/tool/geltool/gadget/kitprinter/proc/toggle(mob/user, silent)
+	if(brute)
+		brute = FALSE
+		playsound(loc, 'sound/machines/computer_typing1.ogg', 5)
+		playsound(loc, 'sound/machines/terminal_on.ogg', 15)  //maybe placehold. maybe it will be left as this. but sure its better than a bland click
+		if(!silent && user)
+			user.visible_message("<span class='notice'>[user] turns [src] to burn mode.</span>",
+		"<span class='notice'>You switch [src] lever to burn mode. <b>Gel gauge reads: [cell.gelquantity]/[cell.gelcapacity]</b></span>")
+		update_kitprinter()
+		return
+
+	if(!brute)
+		brute = TRUE
+		playsound(loc, 'sound/machines/computer_typing1.ogg', 15)
+		playsound(loc, 'sound/machines/terminal_on.ogg', 5 , 1) //maybe placehold. maybe it will be left as this. but sure its better than a bland click
+		if(!silent && user)
+			user.visible_message("<span class='notice'>[user] turns [src] to brute mode.</span>",
+		"<span class='notice'>You switch [src] lever to brute mode. <b>Gel gauge reads: [cell.gelquantity]/[cell.gelcapacity]</b></span>")
+		update_kitprinter()
+		return
+
+/obj/item/tool/geltool/gadget/kitprinter/attack(mob/living/carbon/M as mob, mob/user as mob)
+
+	if(!cell || cell.gelquantity < gelefficiency)
+		to_chat(user, "<span class='warning'>The [src] is out of gel.</span>")
+		return
+
+	if (!ishuman(M))
+		return
+
+	var/mob/living/carbon/human/H = M
+	var/datum/limb/affecting = user.client.prefs.toggles_gameplay & RADIAL_MEDICAL ? radial_medical(H, user) : H.get_limb(user.zone_selected)
+
+	if(!affecting)
+		return TRUE
+
+	if(affecting.surgery_open_stage != 0)
+		to_chat(user, "<span class='warning'>Close the incision before using a [src].</span>")
+		return
+
+	var/bandaged = affecting.bandage()
+	var/disinfected = affecting.disinfect()
+	var/heal_amt = heal_rate
+	var/datum/wound/W = affecting.wounds
```

Comment:
why those vars?

Reply:
bandaged is so it stops infinite bandaging
disinfect is for us to use while closing wounds
heal amt. is so it allows us to modify it later in case we need a more fine tune in balance (for already patched. and for open wounds)

wound/W is. the wound itself

(mostly to improve readablity of the code and cause. in the W specifically we would start to get stuff. like 

``<span class='notice'>[user] applies the printkit on [affecting.wounds.desc] on [M]'s [affecting.display_name].</span>

when we can just [W.desc]

Reply:
that barely improve visibility, these vars are useless IMO